torch_xla/csrc/ir.cpp

```diff
@@ -179,16 +191,17 @@ const xla::Shape& Node::shape(size_t output_index) const {
 void Node::AddOperand(NodePtr node, size_t index) {
```

Comment:
I think we should refined `NodePtr` to be a pointer of base Node class instead of pointer of `xla::Node`. Is there anything blocking us from doing that?

Reply:
Good point, I'll include this in a new commit in this PR.

Reply:
Updated in the latest commit.

Reply:
Yep that is correct.